rca/people/models.py

```diff
@@ -1,65 +1,172 @@
 from django.db import models
+from django.utils.translation import gettext_lazy as _
 from modelcluster.fields import ParentalKey
-from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, MultiFieldPanel
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    ObjectList,
+    PageChooserPanel,
+    StreamFieldPanel,
+    TabbedInterface,
+)
+from wagtail.core.fields import RichTextField, StreamField
 from wagtail.images.edit_handlers import ImageChooserPanel
 
-from rca.utils.models import BasePage
+from rca.utils.blocks import AccordionBlockWithTitle, GalleryBlock, LinkBlock
+from rca.utils.models import BasePage, RelatedPage
 
 
-class SocialMediaProfile(models.Model):
-    person_page = ParentalKey("StaffPage", related_name="social_media_profile")
-    site_titles = (("twitter", "Twitter"), ("linkedin", "LinkedIn"))
-    site_urls = (
-        ("twitter", "https://twitter.com/"),
-        ("linkedin", "https://www.linkedin.com/in/"),
+class AreaOfExpertise(models.Model):
+    title = models.CharField(max_length=128)
+
+    def __str__(self):
+        return self.title
+
+
+class StaffPageAreOfExpertisePlacement(models.Model):
+    page = ParentalKey("StaffPage", related_name="related_area_of_expertise")
+    area_of_expertise = models.ForeignKey(
+        AreaOfExpertise, on_delete=models.CASCADE, related_name="staff"
     )
-    service = models.CharField(max_length=200, choices=site_titles)
-    username = models.CharField(max_length=255)
+    panels = [FieldPanel("area_of_expertise")]
+
 
-    @property
-    def profile_url(self):
-        return dict(self.site_urls)[self.service] + self.username
+class StaffPageRelatedResearchPage(RelatedPage):
+    source_page = ParentalKey("StaffPage", related_name="related_research_centre_pages")
```

Comment:
This is a little bit backwards.... `RelatedPage` has field to define the 'target page', but that's only linking to the `Page` model, meaning you always need to make additional queries to get the specific page. It's really the 'source_page' that should be generic... as it doesn't really matter what type of pages have related pages. Really, this should just be:

```
class RelatedResearchCenterPage(Orderable):
    source_page = ParentalKey("Page", related_name="related_research_centre_pages")
    page = models.ForeignKey("research.ResearchCentrePage")

    panels = [PageChooserPanel("page")]
```

This way, you'll always get the specific `ResearchCenterPage` when you follow the relationship, and `PageChooserPanel` will automatically filter results to pages of that type.

You also DON'T need the specific page in order to get it's related pages... you can get them straight from the vanilla `Page`.

And if any other page type needs to have a list of related research center pages in future, all you need to do is add `InlinePanel('related_research_centre_pages')` to the panels, and you're sorted.

Reply:
Thanks Andy, I've reworks these in https://github.com/torchbox/rca-wagtail-2019/pull/426/commits/e8a19768879dc1d91596a1afe6520276fcb88edf